src/Controller/ProductItem/VarientController.php

```diff
@@ -0,0 +1,121 @@
+        catch(\Exception $e){
+            return $this->json($e->getMessage(), Response::HTTP_BAD_REQUEST);
+        }
+    }
+
+    #[Route('/create/{serial}/denied', name: 'app_varient_create_serial_denied', methods: ['GET'])]
+    public function denied($serial,VarientRepository $varientRepository ,VarientManagement $varientManager){
+        $varientManager->deleteVarient($serial,$varientRepository);
+        return $this->json(
+                ["massage" => "Varient denied successfully",
+                "status" => 200]
+        );
+    }
+
+    #[Route('/create/{serial}/confirm', name: 'app_varient_create_serial_confirm', methods: ['GET'])]
+    public function confirmCreate($serial,VarientRepository $varientRepository,VarientManagement $varientManager): Response
+    {
+        $varientManager->confirmVarient($serial,$varientRepository);
+        return $this->json(
+                ["massage" => "Varient confirmed successfully",
+                "status" => 200]
+        );
+    }
+
+    #[Route('/read/{serial}', name: 'app_varient_read', methods: ['GET'])]
+    public function read($serial,VarientRepository $varientRepository,VarientManagement $varientManager):Response
+    {
+        try {
+            $varient = $varientManager->readVarient($serial,$varientRepository);
+            return $this->json([
+                "data" => $varient,
+                "status" => 200
+            ]);
+        } catch(\Exception $e){
+            return $this->json($e->getMessage(), Response::HTTP_BAD_REQUEST);
+        }
+    }
+
+    #[Route('/update/{serial}', name: 'app_varient_update', methods: ['POST'])]
+    public function update($serial,Request $request,VarientRepository $varientRepository ,VarientManagement $varientManager): Response
+    {
+        $body = $request->toArray();
+        try {
+            $varientManager->updateVarient($serial,$body['quantity'],$varientRepository);
+            return $this->json(
+                  ["massage" => "Varient updated successfully",
+                    "status" => 200]
+            );
+        } catch(\Exception $e){
+            return $this->json($e->getMessage(), Response::HTTP_BAD_REQUEST);
+        }
+    }
+
+    #[Route('/delete/{serial}', name: 'app_varient_delete', methods: ['GET'])]
+    public function delete($serial, VarientRepository $varientRepository ,VarientManagement $varientManager){
+        try {
+            $varientManager->updateVarient($serial, 0, $varientRepository);
+            return $this->json(
+                ["massage" => "Varient deleted successfully",
+                    "status" => 200]
+            );
+        } catch(\Exception $e){
+            return $this->json($e->getMessage(), Response::HTTP_BAD_REQUEST);
+        }
+    }
+    #[Route('/show', name: 'app_varient_show', methods: ['GET'])]
+    public function show(VarientRepository $varientRepository,VarientManagement $varientManager): Response
+    {
+        $filters_eq = array("status" => 1);
```

Comment:
in variant entity define public const STATUS_VALIDATE_SUCCESS;
public const STATUS_VALIDATE_PENDING;